docs/checkstyle/pom.xml

```diff
@@ -0,0 +1,52 @@
+<?xml version="1.0"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <groupId>org.opencastproject</groupId>
+  <artifactId>checkstyle</artifactId>
+  <version>9-SNAPSHOT</version>
```

Comment:
This looks like it could easily be forgotten when doing a version bump.
Why do we need to have code for external projects in here?
Shouldn't they live in their codebase?

Reply:
Why not just copy the configuration? If it's an external project it can have its own rules and this would indirectly promise a stable API for checkstyle rules which get published to Maven Central while it would serve no purpose within this project.

It's more a -0 than a -1 from me but in general, I think that things which are not used by Opencast should not live within Opencast since it confuses people (wtf is this for). But let's see what others think about this.

Reply:
@lkiesow you bring up a few good points I hadn't considered, yet. So I'm actually more of a +0 as opposed to a +1 myself now. :sweat_smile: However, I still think this could be useful. External projects could (for whatever reason) use the Opencast style without having to duplicate the config. Granted, since these rules might just change out from under them, they probably want to pin the version they depend on rather than just always using the latest, and at that point you might say they could also just copy the file, sure. :shrug: 

I also don't think it hurts, though. I personally don't expect any stability from this, beyond the fact that from experience and git history I can tell that the corresponding files don't change that much. :wink: Which is actually more than I can say about the **other** Opencast packages/modules that external projects like the Annotation Tool depend on, which are also published to Central, and which we also don't give any stability guarantees for, as was stated rather explicitly in some technical meeting where I brought up the Annotation Tool breaking because of changes in `common`.

Now, whether or not that's a good thing is a completely different discussion, which also kind of ties in with @gregorydlogan's point: Does Opencast want to support external projects like the Annotation Tool? I personally think it should, because I don't think it scales to pull everything that might have dependencies to Opencast into the main source tree. For one thing, not every project wants to/can evolve at the same speed as Opencast or support as many concurrent versions. That's also why I personally would like to keep the Annotation Tool separate. :thinking: Although I'm also not dead set on that opinion; I also see some advantages of the AT being integrated more tightly, like piggybacking on the existing test/demo infra. Keeping the style of the project consistent with Opencast is useful in any case, though, I think, exactly for the reasons Greg mentioned. Like, the AT doesn't need to live in `opencast/opencast` for committers to contribute; everyone is welcome in our little corner here. :wink: 

As for how to package this: I'm personally a fan of small, targeted artifacts. Who knows what I will pull in if I depend on `org.opencastproject.base` only to get the checkstyle configuration. :thinking: Depends on what else is in it, too, of course. But do we have other things already? It would also seem kind of weird to me if there is **just** the checkstyle config in this generic package. :thinking: Although in the specific case of the Annotation Tool, I might actually want to make `org.opencastproject.base` my parent POM, now that I think about it. (Although I don't actually know whether that works at the moment.) And automatically getting the checkstyle config I want with that actually sounds good. x)

Anyway, for this to work the `packaging` of `base` would need to change. (It's a `pom` now, but needs to be a `jar` to contain resources.) I don't know what implications this might have. :thinking: